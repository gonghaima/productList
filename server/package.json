{
  "name": "docker_web_app",
  "version": "1.0.0",
  "description": "Node.js on Docker",
  "author": "",
  "main": "server.js",
  "scripts": {
    "start": "node server.js"
  },
  "dependencies": {
    "express": "^4.16.1",
    "node-fetch": "^2.3.0"
  }
}
