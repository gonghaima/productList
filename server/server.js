const express = require('express');
const fetch = require('node-fetch');

// Constants
const PORT = 8081;
const HOST = '0.0.0.0';

// App
const app = express();
app.get('/api/product', (req, res) => {
  const apiUrl = 'https://whitechdevs.github.io/reactjs-test/products.json';
  fetch(apiUrl)
    .then(result => result.json())
    .then(data => {
      res.send({ data });
    })
    .catch(err => {
      res.redirect('/error', err);
    });
});

app.listen(PORT, HOST);
