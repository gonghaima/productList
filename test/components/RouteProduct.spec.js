import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import RouteProduct from 'components/RouteProduct';

describe('RouteProduct', () => {
  it('should redirect for all access', () => {
    const render = renderToString(
      <Router initialEntries={['/private']}>
        <RouteProduct exact path="/private" component={() => <div>PRIVATE</div>} />
      </Router>,
    );

    expect(render).toMatchSnapshot();
  });
});
