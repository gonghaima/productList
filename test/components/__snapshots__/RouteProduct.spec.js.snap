// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteProduct should redirect for all access 1`] = `

<div data-reactroot>
  PRIVATE
</div>

`;
